beacon-chain/rpc/validator/status.go

```diff
@@ -93,54 +93,44 @@ func (vs *Server) validatorStatus(ctx context.Context, pubKey []byte, headState
 		resp.ActivationEpoch = int64(val.ActivationEpoch)
 	}
 
-	// If no connection to ETH1, the deposit block number or position in queue cannot be determined.
-	if !vs.Eth1InfoFetcher.IsConnectedToETH1() {
-		log.Warn("Not connected to ETH1. Cannot determine validator ETH1 deposit block number")
-		return resp
-	}
-
-	_, eth1BlockNumBigInt := vs.DepositFetcher.DepositByPubkey(ctx, pubKey)
-	if eth1BlockNumBigInt == nil { // No deposit found in ETH1.
-		return resp
-	}
-
-	if resp.Status == ethpb.ValidatorStatus_UNKNOWN_STATUS {
-		resp.Status = ethpb.ValidatorStatus_DEPOSITED
-	}
-
-	resp.Eth1DepositBlockNumber = eth1BlockNumBigInt.Uint64()
-
-	depositBlockSlot, err := vs.depositBlockSlot(ctx, eth1BlockNumBigInt, headState)
-	if err != nil {
-		return resp
-	}
-	resp.DepositInclusionSlot = int64(depositBlockSlot)
+	switch resp.Status {
+	case ethpb.ValidatorStatus_UNKNOWN_STATUS, ethpb.ValidatorStatus_DEPOSITED:
+		// If no connection to ETH1, the deposit block number or position in queue cannot be determined.
+		if !vs.Eth1InfoFetcher.IsConnectedToETH1() {
+			log.Warn("Not connected to ETH1. Cannot determine validator ETH1 deposit block number")
+			return resp
+		}
 
-	// If validator has been activated at any point, they are not in the queue so we can return
-	// the request early. Also if the validator has exited,slashed or initiated its exit
-	//  we return the request early too. We only proceed if its status is pending active
-	//  Additionally, if idx is zero (default return value) then we know this
-	// validator cannot be in the queue either.
-	if resp.Status != ethpb.ValidatorStatus_PENDING || idx == 0 {
-		return resp
-	}
+		_, eth1BlockNumBigInt := vs.DepositFetcher.DepositByPubkey(ctx, pubKey)
+		if eth1BlockNumBigInt == nil { // No deposit found in ETH1.
+			return resp
+		}
+		resp.Eth1DepositBlockNumber = eth1BlockNumBigInt.Uint64()
 
-	var lastActivatedValidatorIdx uint64
-	for j := headState.NumValidators() - 1; j >= 0; j-- {
-		val, err := headState.ValidatorAtIndex(uint64(j))
+		depositBlockSlot, err := vs.depositBlockSlot(ctx, eth1BlockNumBigInt, headState)
```

Comment:
If a validators status is DEPOSITED, it means their deposit has already been processed into the beacon state, should we show them the estimated DepositInclusionSlot?

Reply:
It would be a nice touch but not critical, can perhaps be a good first issue